lib/errors.ts

```diff
@@ -120,6 +120,20 @@ export class BalenaMalformedToken extends BalenaError {
 }
 BalenaMalformedToken.prototype.code = 'BalenaMalformedToken';
 
+/**
+ * @summary Balena settings permission error
+ * @class
+ * @public
+ *
+ * @param {Error} error - usually an EACCESS error
+ * @return {Error} error instance
+ *
+ * @example
+ * throw new errors.BalenaSettingsPermissionError()
+ */
+export class BalenaSettingsPermissionError extends BalenaError {}
+BalenaSettingsPermissionError.prototype.code = 'BalenaSettingsPermissionError';
```

Comment:
Just wondering about the merits of having this dedicated error class (`BalenaSettingsPermissionError`) instead of simply checking `err.code === 'EACCESS'` in downstream applications like the balena CLI. My doubt builds in part on past experience where `err instanceof BalenaSettingsPermissionError` may fail for `balena-errors` classes, which led me to write this `instanceOf` function in the CLI: https://github.com/balena-io/balena-cli/blob/v12.23.0/lib/errors.ts#L49-L54

Reply:
Personally I prefer the module to throw its own error so that you can do
```ts
try {
	mod.fn() // throws mod.ErrorType
} catch (e) {
	if (e instanceof mod.ErrorType) {
		...
	}
}
```
and it avoids issues with
```ts
try {
	mod.fn() // throws errors@1.0.0.ErrorType
} catch (e) {
	if (e instanceof errors@1.0.1.ErrorType) {
		...
	}
}
```
where despite it throwing an `errors.ErrorType` and you're checking against `errors.ErrorType` they don't actually match due to using different versions of the module (which can easily happen because of the way module resolution works with nested modules, it's even possible to have a tree structured such that both places are using eg `errors@1.0.0` but it's two separate copies that cannot be deduped and will not match via `instanceof`)

cc @thgreasi @iamsolankiamit since you both add to this module fairly often

Reply:
I believe having a specific error for failing to access the settings storage is better than checking for `EACCESS` because the latter is too generic, any other action that handles files on `balena-cli` could result on `EACCESS` and the message would need to include all cases.

The current message for `EACCESS` error, already includes more than one case: https://github.com/balena-io/balena-cli/blob/v12.23.0/lib/errors.ts#L121

Reply:
FYI: https://github.com/balena-io-modules/balena-errors/pull/48/files

Reply:
@pdcastro I created [this draft](https://github.com/balena-io-modules/balena-settings-storage/pull/31) to discuss how to have errors on each project after `balena-errors` deprecation.